airflow/lineage/hook.py

```diff
@@ -159,6 +160,9 @@ def add_input_asset(
         asset_extra: dict | None = None,
     ):
         """Add the input asset and its corresponding hook execution context to the collector."""
+        if len(self._inputs) >= 100:
```

Comment:
Should we extract this `100` as a variable?

Reply:
Done.